pandas/tests/frame/test_constructors.py

```diff
@@ -723,6 +724,18 @@ def test_constructor_period(self):
         assert df["a"].dtype == a.dtype
         assert df["b"].dtype == b.dtype
 
+    def test_constructor_period_data(self):
+        # GH 34832
+        data = pd.Period("2012-01", freq="M")
+        df = DataFrame(index=[0, 1], columns=["a", "b"], data=data)
+
+        assert df["a"].dtype == PeriodDtype("M")
+        assert df["b"].dtype == PeriodDtype("M")
+
+        expected = DataFrame({"a": [data] * 2, "b": [data] * 2})
```

Comment:
```suggestion
        arr = pd.array([data] * 2, dtype=PeriodDtype("M")
        expected = DataFrame({"a": arr, "b": arr})
```

(this should give a bit more confidence in the test IMO, as then the construction of the expected result doesn't rely on inference of (a list of) scalars as well)